site-builder/src/site/resource.rs

```diff
@@ -384,8 +452,7 @@ impl ResourceManager {
         Ok(Some(Resource::new(
             full_path_to_resource_path(full_path, root)?,
             full_path.to_owned(),
-            content_type,
-            *content_encoding,
+            http_headers.unwrap(),
```

Comment:
Important: use `except("explanation why bad stuff should not happen")` instead of `unwrap`
```suggestion
            http_headers.except("we have created the headers above"),
```

Nit: removing TryFrom also removes the need for an expect here.